TaskList/app/controllers/tasks_controller.rb

```diff
@@ -1,23 +1,27 @@
 class TasksController < ApplicationController
+  before_action :get_task, only: [:show, :edit, :update, :toggle_completed]
+
   def index
-    @title = "Task List"
     @tasks = Task.all
   end
 
   def show
-    @title = "About This Task"
-    id = params[:id]
-    @tasks = Task.where(id: id)
+    @tasks = [@task]
   end
 
   def new
-    @title = "Add a Task"
     @task = Task.new()
   end
 
   def create
-    Task.create(task_params[:task])
-    redirect_to('/')
+    @action_url = "/tasks/"
+    @method = :post
+    @task = Task.new(task_params[:task])
+    if @task.save
```

Comment:
Awesome error checking for `save`! We haven't gotten to ActiveRecord validations yet, but this is generally exactly what you want to do. Verify that the save worked, and if not re-display the same form but with the original form values filled in again.